test/extended/operators/check_crd_status.go

```diff
@@ -0,0 +1,115 @@
+package operators
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// check_crds returns a list of names of CRDs that don't have subresource.status
+// For now, it ignores the ones that are currently failing.
+func check_crds(crd_item_list []v1.CustomResourceDefinition) []string {
+
+	contains := func(str_list []string, name string) bool {
+		for _, item := range str_list {
+			if name == item {
+				return true
+			}
+		}
+		return false
+	}
+
+	// These 17 CRDs do not have subresource.status.
+	//
+	exceptions_list := []string{
+		"clusternetworks.network.openshift.io",
+		"consoleplugins.console.openshift.io",
+		"consolequickstarts.console.openshift.io",
+		"consoleyamlsamples.console.openshift.io",
+		"egressnetworkpolicies.network.openshift.io",
+		"hostsubnets.network.openshift.io",
+		"machineconfigs.machineconfiguration.openshift.io",
+		"netnamespaces.network.openshift.io",
+		"networks.config.openshift.io",
+		"networks.operator.openshift.io",
+		"nodes.config.openshift.io",
+		"operatorpkis.network.operator.openshift.io",
+		"profiles.tuned.openshift.io",
+		"rangeallocations.security.internal.openshift.io",
+		"rolebindingrestrictions.authorization.openshift.io",
+		"securitycontextconstraints.security.openshift.io",
+		"tuneds.tuned.openshift.io",
+	}
+
+	failures := []string{}
+	for _, crd_item := range crd_item_list {
+
+		// This test is interested only in CRDs that end with "openshift.io".
+		if !strings.HasSuffix(crd_item.ObjectMeta.Name, "openshift.io") {
+			continue
+		}
+
+		// Use the latest version
+		index := len(crd_item.Spec.Versions) - 1
```

Comment:
Is the latest version always the last in the list?

Reply:
Hm... I looked at the version `name` field and it can be something like `v1`, `v1beta`, etc. and so we can't tell which one is latest.  So I added a loop to check all versions.  It turns out today, there is only one version in all of the CRDs ending in `openshift.io`.

Reply:
Ah good, that makes sense, I recall it was a pretty huge pain to support more than one version at a time.